pmd-core/src/test/java/net/sourceforge/pmd/ReportTest.java

```diff
@@ -168,6 +169,48 @@ public void testTreeIterator() {
         assertEquals(2, treeCount);
     }
 
+    @Test
+    public void testFilterViolations() {
+        Report r = new Report();
+        RuleContext ctx = new RuleContext();
+        Rule rule = new MockRule("name", "desc", "msg", "rulesetname");
+        Node node1 = getNode(5, 5);
+        r.addRuleViolation(new ParametricRuleViolation<>(rule, ctx, node1, rule.getMessage()));
+        Node node2 = getNode(5, 6);
+        r.addRuleViolation(new ParametricRuleViolation<>(rule, ctx, node2, "to be filtered"));
+
+        Report filtered = r.filterViolations(new Predicate<RuleViolation>() {
+            @Override
+            public boolean test(RuleViolation ruleViolation) {
+                return !("to be filtered".equals(ruleViolation.getDescription()));
```

Comment:
Unnecessary parens (the pmd7 rule probably would complain)
```suggestion
                return !"to be filtered".equals(ruleViolation.getDescription());
```